File: life.rb
COL,ROW=20,20
def init;Array.new(COL){Array.new(ROW){rand(2)}}end
def neig(c,x,y);[0,1,-1].product([0,1,-1]).drop(1).collect{|i,j|c[(x+i+COL)%COL][(y+j+ROW)%ROW]}.inject(:+)end
def nexT(c);Array.new(COL){|i|Array.new(ROW){|j|neig(c,i,j)==3 ? 1 : (neig(c,i,j)==2 ? c[i][j] : 0)}}end
def prin(c);puts c.collect{|v|v.collect{|x|x==1 ? "[]" : "  "}.join("")}.join("\n")end
def life(c=init);prin(c);puts Array.new(ROW){"--"}.join("");sleep 1;life(nexT(c))end
life

File: life.py
import random,time
COL,ROW=30,30
init=lambda:[[random.randint(0,1) for j in range(ROW)] for i in range(COL)]
neig=lambda c,x,y:sum([c[(x+i+COL)%COL][(y+j+ROW)%ROW] for j in [0,1,-1] for i in [0,1,-1]][1:])
nexT=lambda c:[[1 if neig(c,i,j)==3 else (c[i][j] if neig(c,i,j)==2 else 0) for j in range(ROW)] for i in range(COL)]
prin=lambda c:print('\n'.join([''.join(['[]' if c[i][j]==1 else '  ' for j in range(ROW)]) for i in range(COL)]))
life=lambda c=init():prin(c) or print(''.join(['--' for i in range(ROW)])) or time.sleep(1) or life(nexT(c))
life()
